application/main.py

```diff
@@ -573,7 +572,92 @@ def __eq__(self, other):
                (self.validation_results == other.validation_results)
 
 
+@application.route('/build_implementer_agreement_results/<id>')
+@login_required
+def build_implementer_agreement_results(user_data, id):
+    with database.Session() as session:
+
+        model = session.query(database.model).filter(database.model.code == id).all()[0]
+        tasks = {t.task_type: t for t in model.tasks}
+        instances = {i: i for i in model.instances}
+
+        imp_agreement_results = [imp_ag.serialize() for imp_ag in tasks['implementer_agreements_task'].results]
+        instances_results = [ins.serialize() for ins in instances]
```

Comment:
Maybe another way of doing it is to get the dictionary values instead of calling `serialize(`) on each instance
` instances_results = [ins.serialize() for ins in instances.values()]`

That way, it seems 'instances' is reusable for something else. For example
`instances = {i.id: i for i in model.instances}` #or global_id for example

Reply:
Done :)